gapic-generator/templates/default/layouts/_ruby.erb

```diff
@@ -1,13 +1,18 @@
+<%- assert_locals namespaces -%>
 # frozen_string_literal: true
 
 <%= render partial: "shared/license" %>
-
+<% if @requires %>
 <%= @requires %>
-
-<% namespaces.each do |namespace| %>
-module <%= namespace.upcase_first %>
-<% end %>
-<%= yield %>
-<% namespaces.count.times do %>
-end
 <% end %>
+<%- namespaces.each_with_index do |n, i| -%>
+<%= indent "module #{n.upcase_first}", i*2 %>
```

Comment:
Will this work? 

It needs to be able to handle overrides via the `ruby_package` (example: https://github.com/googleapis/googleapis/blob/master/google/pubsub/v1/pubsub.proto)

Feel free to open an issue and defer it for this PR - this caused a lot of pain the current generator so I don't want to let it get out of hand again.

Reply:
Yep, I can have the api schema determine the address from the ruby_package option. Just haven't gotten around to it.